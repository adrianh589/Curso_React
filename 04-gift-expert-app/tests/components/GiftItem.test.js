import {screen, render} from "@testing-library/react";
import {GiftItem} from "../../src/components/index.js";

describe('Pruebas en <GiftItem>', () => {

    const title = 'Saitama';
    const url = 'www.saitama.com'

    test('Debe de hacer match con el snapshot', () => {
       const {container} = render(<GiftItem title={title} url={url}/>);
       expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el url y el alt indicado', () =>{
        render(<GiftItem title={title} url={url}/>);
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    });

    test('Debe de mostrar el titulo en el componente', () => {
       render(<GiftItem title={title} url={url} /> );
       expect(screen.getByText(title)).toBeTruthy();
    });
});