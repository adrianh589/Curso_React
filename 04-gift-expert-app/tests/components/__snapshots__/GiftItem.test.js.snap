// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="www.saitama.com"
    />
  </div>
</div>
`;
